From 62756ccf38044c4c042d800be7b46e031ef95234 Mon Sep 17 00:00:00 2001
From: "sw.multimedia" <sw.multimedia@starfivetech.com>
Date: Mon, 25 Apr 2022 19:22:45 +0800
Subject: [PATCH] ffmepg add bsf mp4toannexb to fix decoder mp4/mkv issue

---
 libavcodec/omxdec.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavcodec/omxdec.c b/libavcodec/omxdec.c
index 0ccc1dc..7bb1d27 100755
--- a/libavcodec/omxdec.c
+++ b/libavcodec/omxdec.c
@@ -931,6 +931,7 @@ AVCodec ff_h264_omx_decoder = {
     .capabilities     = AV_CODEC_CAP_DELAY,
     .caps_internal    = FF_CODEC_CAP_INIT_THREADSAFE | FF_CODEC_CAP_INIT_CLEANUP,
     .priv_class       = &omx_h264dec_class,
+    .bsfs             = "h264_mp4toannexb",
 };
 
 static const AVClass omx_hevcdec_class = {
@@ -952,6 +953,7 @@ AVCodec ff_hevc_omx_decoder = {
     .capabilities     = AV_CODEC_CAP_DELAY,
     .caps_internal    = FF_CODEC_CAP_INIT_THREADSAFE | FF_CODEC_CAP_INIT_CLEANUP,
     .priv_class       = &omx_hevcdec_class,
+    .bsfs             = "hevc_mp4toannexb",
 };
 
 static const AVClass omx_mjpegdec_class = {
-- 
2.25.1

