From 2faf0e97fa4f1fdf8a2d22407966ea8747581eff Mon Sep 17 00:00:00 2001
From: "zejian.su" <zejian.su@starfivetech.com>
Date: Mon, 30 Oct 2023 17:35:05 +0800
Subject: [PATCH] Let the option: width, height work again.

Signed-off-by: zejian.su <zejian.su@starfivetech.com>
---
 core/libcamera_app.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/libcamera_app.cpp b/core/libcamera_app.cpp
index cc76e9b..a49e31d 100644
--- a/core/libcamera_app.cpp
+++ b/core/libcamera_app.cpp
@@ -252,6 +252,10 @@ void LibcameraApp::ConfigureViewfinder()
 	//auto area = camera_->properties().get(properties::PixelArrayActiveAreas);
 	if (options_->viewfinder_width && options_->viewfinder_height)
 		size = Size(options_->viewfinder_width, options_->viewfinder_height);
+	else if (options_->width && options_->height)
+	{
+		size = Size(options_->width, options_->height);
+	}
 	//else if (area)
 	//{
 	//	// The idea here is that most sensors will have a 2x2 binned mode that
-- 
2.34.1

