Clear all framebuffer content when mapping.

Signed-off-by: Windsome Zeng <windsome.zeng@starfivetech.com>

diff -purN a/sys/kms/gstkmsallocator.c b/sys/kms/gstkmsallocator.c
--- a/sys/kms/gstkmsallocator.c	2022-08-31 10:57:31.759791991 +0800
+++ b/sys/kms/gstkmsallocator.c	2022-08-31 09:25:30.132182006 +0800
@@ -384,6 +384,9 @@ gst_kms_memory_map (GstMemory * mem, gsi
   }
   kmsmem->bo->ptr = out;
 
+  /* clear the framebuffer to 0 */
+   memset(out, 0, kmsmem->bo->size);
+
 out:
   g_atomic_int_inc (&kmsmem->bo->refs);
   return kmsmem->bo->ptr;
